__index__.md

```diff
@@ -1,8 +1,10 @@
 
 - **[cache/v0.1](/cache/v0.1)** ([📄 graphql](cache/v0.1/cache-v0.1.graphql))
+- **[cache/v0.2](/cache/v0.2)** ([📄 graphql](cache/v0.2/cache-v0.2.graphql))
 - **[core/v0.1](/core/v0.1)** ([📄 graphql](core/v0.1/core-v0.1.graphql))
 - **[core/v0.2](/core/v0.2)** ([📄 graphql](core/v0.2/core-v0.2.graphql))
 - **[cost/v0.1](/cost/v0.1)** ([📄 graphql](cost/v0.1/cost-v0.1.graphql))
+- **[fakes/v0.0](/fakes/v0.0)** ([📄 graphql](fakes/v0.0/fakes-v0.0.graphql))
```

Comment:
I think we should be considering this as `mock`, not `fake` since most users will understand what we mean by "mocks" instead of "fake".

Reply:
Naming is the hardest thing... I considered mock for a while but coming from a JVM/Android world, "mocking" is pretty much [seen as an anti-pattern](https://www.reddit.com/r/Kotlin/comments/fovn9z/mocking_is_not_practical_use_fakes/) these days, mostly because of mockito and so on. I agree this is very Android specific though.

The thing that tipped me over is that there are lots of connections to existing tools in the ecosystem:

* [graphql-faker](https://github.com/graphql-kit/graphql-faker)
* [faker.js](https://fakerjs.dev/)
* [datafaker](https://github.com/datafaker-net/datafaker)

Reply:
I can see that. 

I think there's a piece here which is - if we're faking the data (where `fake` is probably the right term) to return back to users, are we not just effectively mocking the subgraph responses? 

There's probably an argument either way, just that when I see `@fake` I assume the field itself is fake/wrong/invalid/etc, not that it can return X fake data. 

Also as a meta question to answer in another location- do we expect to have some default faking logic (e.g. for strings we return `Lorem ipsum` or do we return nothing?)?

Reply:
Re naming: I hear you. It's just that graphql-faker has had some success with `@fake` and it seems this is where the legacy lies and there hasn't been a strong push to rename. Maybe @IvanGoncharov will have more insights there?

Re default logic: I would say this is outside the directives. apollo-kotlin-faker returns `Lorem ipsum` but this is implementation dependent. It doesn't have to be specified.

Reply:
Okay taking a step back- is this directive for clients to use or for the router to use to fake the data?

Reply:
It's for whoever needs it. My own needs is for clients to use in tests/early development phases but I see nothing wrong in the router using them as well. I think it's the same use case ultimately?

Reply:
I would push us to support this in the router instead, since if the router did handle this we could do some fun things like add support in Rhai or coprocessor to do custom faking of data :) 

I think either's fine, we just need to be crystal clear on what the expected DX is for consumers; aka, how does this differ from a comment indicating the type? And if it's automation, ensuring we have adequate automation around it. 

Pushing into the router means this work is more accessible since there's no custom tooling needed per-lib.

Reply:
Having support in the router would be pretty dope!